File: _includes/code/howto/indexes/indexes-inverted-v3.py
# TODO: Configure as part of the test harness

class_name = "ConfigCollection"

########################
### CLIENT CONNECTION ##
########################

import os
import weaviate

client = weaviate.Client(
    url="http://localhost:8080",
    additional_headers={"X-Cohere-Api-Key": os.getenv("COHERE_API_KEY")},
)

##############
### SEARCH ###
##############

# Delete data from prior runs
if client.schema.exists(class_name):
    client.schema.delete_class(class_name)

# START SearchIndex
class_obj = {
    "class": class_name,
    "vectorizer": "text2vec-huggingface",
    "properties": [
        {
            "name": "title",
            "dataType": ["text"],
            "indexSearchable": False,
        },
}

client.schema.create_class(class_obj)
# END SearchIndex

class_response = client.schema.get()
schema_response = client.schema.get(class_name)

classes = []
for c in class_response["classes"]:
    classes.append(c["class"])
assert class_name in classes, "Class missing"


##############
### FILTER ###
##############

# Delete data from prior runs
if client.schema.exists(class_name):
    client.schema.delete_class(class_name)

# START FilterIndex
class_obj = {
    "class": class_name,
    "vectorizer": "text2vec-huggingface",
    "properties": [
        {
            "name": "title",
            "dataType": ["text"],
            "indexFilterable": False,
        },
}

client.schema.create_class(class_obj)
# END FilterIndex

class_response = client.schema.get()
schema_response = client.schema.get(class_name)

classes = []
for c in class_response["classes"]:
    classes.append(c["class"])
assert class_name in classes, "Class missing"


#############
### RANGE ###
#############

# Delete data from prior runs
if client.schema.exists(class_name):
    client.schema.delete_class(class_name)

# START RangeIndex
class_obj = {
    "class": class_name,
    "vectorizer": "text2vec-huggingface",
    "properties": [
        {
            "name": "scores",
            "dataType": ["int"],
            "indexRangeFilters": True,
        },
    ],
}

client.schema.create_class(class_obj)
# END RangeIndex

class_response = client.schema.get()
schema_response = client.schema.get(class_name)

classes = []
for c in class_response["classes"]:
    classes.append(c["class"])
assert class_name in classes, "Class missing"

############
### BM25 ###
############

# Delete data from prior runs
if client.schema.exists(class_name):
    client.schema.delete_class(class_name)

# START BM25Index
class_obj = {
    "class": class_name,
    "invertedIndexConfig": {
        "bm25": {
            "b": 0.7,
            "k1": 1.25
        },
    }
}

client.schema.create_class(class_obj)
# END BM25Index

class_response = client.schema.get()
schema_response = client.schema.get(class_name)

classes = []
for c in class_response["classes"]:
    classes.append(c["class"])
assert class_name in classes, "Class missing"

########################
### COLLECTION LEVEL ###
########################

# Delete data from prior runs
if client.schema.exists(class_name):
    client.schema.delete_class(class_name)

# START CollLevIndex
class_obj = {
    "class": class_name,
    "invertedIndexConfig": {
        "indexTimestamps": True,
        "indexNullState": True,
        "indexPropertyLength": True
    }
}

client.schema.create_class(class_obj)
# END CollLevIndex

class_response = client.schema.get()
schema_response = client.schema.get(class_name)

classes = []
for c in class_response["classes"]:
    classes.append(c["class"])
assert class_name in classes, "Class missing"

File: _includes/code/howto/indexes/indexes-inverted-v3.ts
// TODO: Configure as part of the test harness
// TODO: Needs tests

// Imports
import weaviate, { dataType, WeaviateClient,vectorizer, configure } from 'weaviate-client';

// Delete pre-existing collections
async function deleteCollection(client: WeaviateClient, collectionName: string){
 if(client.collections.exists(collectionName)){
   await client.collections.delete(collectionName)
  }
}

// Create client connection
async function getClient(){
  const client: WeaviateClient = weaviate.connectToLocal();

  return client;
}

////////////
// SEARCH //
////////////

// START SearchIndex
// Add this import line
// import { dataType } from 'weaviate-client';

async function searchInvertedCollection(client: WeaviateClient, collectionName: string){
  await client.collections.create({
    name: collectionName,
    properties: [
      {
        name: 'TextProperty',
        dataType: dataType.TEXT,
        indexSearchable: true,
      },
    ],
  })
}
// END SearchIndex

////////////
// FILTER //
////////////

// START FilterIndex
// Add this import line
// import { dataType } from 'weaviate-client';

async function filterInvertedCollection(client: WeaviateClient, collectionName: string){
 await client.collections.create({
   name: collectionName,
   properties: [
     {
       name: 'TextProperty',
       dataType: dataType.TEXT,
       indexFilterable: true,
     },
   ],
 })
}
// END FilterIndex

///////////
// RANGE //
///////////

// START RangeIndex
// Add this import line
// import { dataType } from 'weaviate-client';

async function rangeInvertedCollection(client: WeaviateClient, collectionName: string){
 await client.collections.create({
   name: collectionName,
   properties: [
     {
       name: 'NumericProperty',
       dataType: dataType.INT,
       indexRangeFilters: true,
     },
   ],
 })
}
// END RangeIndex

//////////
// BM25 //
//////////

// START BM25Index
// Add this import line
// import { dataType } from 'weaviate-client';

async function bm25InvertedCollection(client: WeaviateClient, collectionName: string){
 await client.collections.create({
   name: collectionName,
   invertedIndex: {
     bm25: {
       b: 0.7,
       k1: 1.25
     },
   }
 })
}
// END BM25Index

//////////////////////
// COLLECTION LEVEL //
//////////////////////

// START CollLevIndex
// Add this import line
// import { dataType } from 'weaviate-client';

async function collLevInvertedCollection(client: WeaviateClient, collectionName: string){
 await client.collections.create({
   name: collectionName,
   invertedIndex: {
     indexNullState: true,
     indexPropertyLength: true,
     indexTimestamps: true
   }
 })
}
// END CollLevIndex

/////////////////////////////
/// AVOID TOP LEVEL AWAIT ///
/////////////////////////////

// Main
async function main(){
  const collectionName = "ConfigCollection";

  const client = await getClient();

  // Run search code
  await deleteCollection(client, collectionName)
  await searchInvertedCollection(client, collectionName);

  // // Run filter code
  // await deleteCollection(client, collectionName)
  // await filterInvertedCollection(client, collectionName);

  // // Run range code
  // await deleteCollection(client, collectionName)
  // await rangeInvertedCollection(client, collectionName);

  // // Run bm25 code
  // await deleteCollection(client, collectionName)
  // await bm25InvertedCollection(client, collectionName);

  // // Run collection level code
  // await deleteCollection(client, collectionName)
  // await collLevInvertedCollection(client, collectionName);


}

main()
